Rock/Security/Authentication/Google.cs

```diff
@@ -0,0 +1,363 @@
+        /// </summary>
+        /// <value>
+        /// <c>true</c> if [supports change password]; otherwise, <c>false</c>.
+        /// </value>
+        public override bool SupportsChangePassword
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Changes the password.
+        /// </summary>
+        /// <param name="user">The user.</param>
+        /// <param name="oldPassword">The old password.</param>
+        /// <param name="newPassword">The new password.</param>
+        /// <param name="warningMessage">The warning message.</param>
+        /// <returns></returns>
+        /// <exception cref="System.NotImplementedException"></exception>
+        public override bool ChangePassword( UserLogin user, string oldPassword, string newPassword, out string warningMessage )
+        {
+            warningMessage = "not supported";
+            return false;
+        }
+
+        /// <summary>
+        /// Google User Object
+        /// </summary>
+        public class GoogleUser
+        {
+            public string family_name { get; set; }
+            public string name { get; set; }
+            public string picture { get; set; }
+            public string locale { get; set; }
+            public string gender { get; set; }
+            public string email { get; set; }
+            public string link { get; set; }
+            public string given_name { get; set; }
+            public string id { get; set; }
+            public string hd { get; set; }
+            public bool verified_email { get; set; }
+        }
+
+        /// <summary>
+        /// Gets the name of the Google user.
+        /// </summary>
+        /// <param name="googleUser">The Google user.</param>
+        /// <param name="accessToken">The access token.</param>
+        /// <returns></returns>
+        public static string GetGoogleUser( GoogleUser googleUser, string accessToken = "" )
+        {
+            string username = string.Empty;
+            string googleId = googleUser.id;
+            string googleLink = googleUser.link;
+
+            string userName = "Google_" + googleId;
+            UserLogin user = null;
+
+            using (var rockContext = new RockContext() )
+            {
+
+                // Query for an existing user 
+                var userLoginService = new UserLoginService(rockContext);
+                user = userLoginService.GetByUserName(userName);
+
+                // If no user was found, see if we can find a match in the person table
+                if ( user == null )
+                    {
+                    // Get name/email from Google login
+                    string lastName = googleUser.family_name.ToString();
+                    string firstName = googleUser.given_name.ToString();
+                    string email = string.Empty;
+                    try { email = googleUser.email.ToString(); }
+                    catch { }
```

Comment:
Yep, I looked at the Facebook auth and you were just following the pattern.  Sorry about that :)  

@azturner Should the Facebook auth be updated to use ToStringSafe instead of the generic try/catch?